nucypher/utilities/events.py

```diff
@@ -378,45 +374,54 @@ def _fetch_events_for_all_contracts(
     if from_block is None:
         raise TypeError("Missing mandatory keyword argument to get_logs: fromBlock")
 
-    # Currently no way to poke this using a public Web3.py API.
-    # This will return raw underlying ABI JSON object for the event
-    abi = event._get_event_abi()
-
     # Depending on the Solidity version used to compile
     # the contract that uses the ABI,
     # it might have Solidity ABI encoding v1 or v2.
     # We just assume the default that you set on Web3 object here.
     # More information here https://eth-abi.readthedocs.io/en/latest/index.html
     codec: ABICodec = web3.codec
 
-    # Here we need to poke a bit into Web3 internals, as this
-    # functionality is not exposed by default.
-    # Construct JSON-RPC raw filter presentation based on human readable Python descriptions
-    # Namely, convert event names to their keccak signatures
-    # More information here:
-    # https://github.com/ethereum/web3.py/blob/e176ce0793dafdd0573acc8d4b76425b6eb604ca/web3/_utils/filters.py#L71
-    data_filter_set, event_filter_params = construct_event_filter_params(
-        abi,
-        codec,
-        address=argument_filters.get("address"),
-        argument_filters=argument_filters,
-        fromBlock=from_block,
-        toBlock=to_block
+    topics = set()
+    event_topics_to_abis = {}
+    for event_type in events:
+        event_abi = event_type._get_event_abi()
+        event_topic = encode_hex(event_abi_to_log_topic(event_abi))  # type: ignore
+        topics.add(event_topic)
+        event_topics_to_abis[event_topic] = event_abi
+
+    event_filter_params = {
+        "address": contract.address,
+        "topics": [list(topics)],
```

Comment:
```suggestion
        "topics": list(topics),
```
This should be a list of topics, right?

Reply:
No. It is a list of list on purpose. See https://web3py.readthedocs.io/en/stable/filters.html#event-log-filters.

For searching for events using topics the first element is the set of topics, and subsequent entries are arguments to filter the events by.

In our case we want multiple topics (events) to be searched for at the same time with no arguments (essentially don't filter on any arguments) so:
`[[A, B, C]]` means search for events with topics A OR B OR C

`[A, B, C]` would mean search for events with topic A, and arguments B, C.